### DISCUSSION TITLE
What is the best configuration for my hardware?

### DISCUSSION THREAD

**Discussion by Dstack86** [madMAx43v3r/chia-plotter#373] [Q&A]
I'm about ready to admit defeat. I need a Mentor!!!

I have 3 dell power edge r610 servers with dual Xeon CPUs (24 cores) and 190GB of ram. I have several SAS 1TB drives and 1 SAS 3.8TB SSD and I'm still getting plot times over 2.5 hrs. when using the SSD for temp1 and a Ram drive for Temp2 with all 24 cores. For some reason, the plot also takes another hr and a half just to copy to the final directory. (SAS 1TB drive)

Questions:

-should I use My 3.8TB SSD for temp1 or do a Raid0 of several 7K RPM drives? it seems like people using just hard disks are getting way better plot times than me even though I'm using a Ram drive and an SSD.

-I noticed my "swap" file is completely maxed out during plotting and when this happens my CPU usage drops to 20-30% I assume this is a bottleneck. If so what is the best way to get around this issue?

-This may be a noob question but if I have installed the Plotter through the terminal how can I update to the latest version? or is it automatic? (I have only been using Linux for a week take it easy on me haha)

If I can get all my hardware configured properly I think Icould be plotting almost 5-6 TB/day. 
If someone was willing to talk with me on discord and help optimize these plotting machines, I would really appreciate it. Im genuinely intrigued by Chia and am trying to grow my Linux and crypto knowledge! Thank you!

**Comment by SebMoore** [comment]
> I'm about ready to admit defeat. I need a Mentor!!!
> 
> I have 3 dell power edge r610 servers with dual Xeon CPUs (24 cores) and 190GB of ram. I have several SAS 1TB drives and 1 SAS 3.8TB SSD and I'm still getting plot times over 2.5 hrs. when using the SSD for temp1 and a Ram drive for Temp2 with all 24 cores. For some reason, the plot also takes another hr and a half just to copy to the final directory. (SAS 1TB drive)
> 
> Questions:
> 
> -should I use My 3.8TB SSD for temp1 or do a Raid0 of several 7K RPM drives? it seems like people using just hard disks are getting way better plot times than me even though I'm using a Ram drive and an SSD.
> 
> -I noticed my "swap" file is completely maxed out during plotting and when this happens my CPU usage drops to 20-30% I assume this is a bottleneck. If so what is the best way to get around this issue?
> 
> -This may be a noob question but if I have installed the Plotter through the terminal how can I update to the latest version? or is it automatic? (I have only been using Linux for a week take it easy on me haha)
> 
> If I can get all my hardware configured properly I think Icould be plotting almost 5-6 TB/day.
> If someone was willing to talk with me on discord and help optimize these plotting machines, I would really appreciate it. Im genuinely intrigued by Chia and am trying to grow my Linux and crypto knowledge! Thank you!

Ok - 
I've got no idea about the 1st question.
2nd question: if you are running a ramdisk for your Temp2, then that shouldn't be a problem - just make sure that your ramdisk is over 110GB or else it will start swapping to your SSD which is bad.
3rd question - there is no update mechanism. You have to delete and then reinstall the entire project.

**Reply by Dstack86** [reply]
Ok so my RAM disk is exactly 110gb should I bump it up?

**Reply by SebMoore** [reply]
Hmm - 110gb should be fine - try reducing swappiness to maybe 10 or so - google this. Also make sure you make your ramdisk with size=110G not 110g (case-sensitive) Apart from that, I'm not really sure what else you could do there.

**Reply by kentgran** [reply]
3rd question - Just do a `git pull` and build again

**Reply by 7by24** [reply]
you may need a high speed ssd for temp dir 1 and save the final files to the 3.6TB ssd, then transfer plot files to your harvesters.
and you can compile the newest version on WSL and copy it to your plotter.
